Firebase/InAppMessaging/Flows/FIRIAMDisplayExecutor.m

```diff
@@ -368,6 +372,52 @@ - (void)checkAndDisplayNextContextualMessageForAnalyticsEvent:(NSString *)eventN
   }
 }
 
+- (FIRInAppMessagingCardDisplay *)
+    cardMessageWithMessageDefinition:(FIRIAMMessageDefinition *)definition
+                   portraitImageData:(FIRInAppMessagingImageData *)portraitImageData
+                  landscapeImageData:(nullable FIRInAppMessagingImageData *)landscapeImageData
+                         triggerType:(FIRInAppMessagingDisplayTriggerType)triggerType {
+  // For easier reference in this method.
+  FIRIAMMessageRenderData *renderData = definition.renderData;
+
+  NSString *title = renderData.contentData.titleText;
+  NSString *body = renderData.contentData.bodyText;
+
+  FIRInAppMessagingActionButton *primaryActionButton = nil;
+  if (definition.renderData.contentData.actionButtonText) {
+    primaryActionButton = [[FIRInAppMessagingActionButton alloc]
+        initWithButtonText:renderData.contentData.actionButtonText
+           buttonTextColor:renderData.renderingEffectSettings.btnTextColor
+           backgroundColor:renderData.renderingEffectSettings.btnBGColor];
+  }
+
+  FIRInAppMessagingActionButton *secondaryActionButton = nil;
+  if (definition.renderData.contentData.secondaryActionButtonText) {
+    secondaryActionButton = [[FIRInAppMessagingActionButton alloc]
+        initWithButtonText:renderData.contentData.secondaryActionButtonText
+           buttonTextColor:renderData.renderingEffectSettings.secondaryActionBtnTextColor
+           backgroundColor:renderData.renderingEffectSettings.btnBGColor];
+  }
+
+  FIRInAppMessagingCardDisplay *cardMessage = [[FIRInAppMessagingCardDisplay alloc]
+          initWithMessageID:renderData.messageID
```

Comment:
This initializer has 14 parameters which is a bit much. 5 are `nullable` - can we break that down and have these properties as settable instead? I still think 9 is too much but I think that's a much larger change to be had.